### ISSUE TITLE
Going up stairs from a basement teleport to different set of stairs.

### ISSUE THREAD

**Issue by UndeadMoose** [CleverRaven/Cataclysm-DDA#36833]
# Describe the bug

Going up stairs from a basement will cause the game to teleport me to a different set of stairs I didn't use to get down in the first place

# Steps To Reproduce

1. Find a house with a basement
2. Go down stairs
3. Go back up

# Expected behavior

Game should teleport me to the same set of stairs I used to get down

# Screenshots

![Screenshot (600)](https://user-images.githubusercontent.com/56947231/72028284-496c4580-32c5-11ea-9e6f-595c881b9079.png)
![Screenshot (601)](https://user-images.githubusercontent.com/56947231/72028293-525d1700-32c5-11ea-889e-034f97c4b565.png)

# Versions and configuration

 - OS: Windows 10
 - Game Version: 0.D-11094-gbb7e3c4
 - Graphics version: Tiles
 - Mods loaded: 
Aftershock
Whaley's Hair Mod
Ben's GF Recipes
Garden Pots
Makeshift Items Mod
More Survival Tools
Hydroponics
Mutant NPCs
Mythical Martial Arts
Folding Parts Pack
Vehicle Additions Pack
Roadheader and other mining vehicles
Tanks and Other Vehicles
Classes and Scenarios Mod
Classic Roguelike Classes
Fuji's Military Professions Pack
Speedy Dex
Sleep Deprivation
Dino Mod
More Locations
Fuji's More Buildings
Urban Development
# Additional context

[Raoul.zip](https://github.com/CleverRaven/Cataclysm-DDA/files/4038041/Raoul.zip)

**Comment by wapcaplet** [comment]
This happens to me in any house with two sets of stairs to the basement. Two locations lead down, but from the basement, the stairs only lead up to one of the two locations.

I can also teleport by peeking - if I use `X` to peek down the stairs, I may be teleported across the house to the other set of down stairs.

![stair-alignment-peek-teleport](https://user-images.githubusercontent.com/238395/72231014-2f0cc300-3576-11ea-935f-74806a21dc1c.png)

- OS: Linux
    - OS Version: LSB Version: core-9.20170808ubuntu1-noarch:security-9.20170808ubuntu1-noarch; Distributor ID: Ubuntu; Description: Ubuntu 18.04.3 LTS; Release: 18.04; Codename: bionic; 
- Game Version: 0.D-11170-g64b5e04a6b [64-bit]
- Graphics Version: Tiles
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    No Rail Stations [No_Rail_Stations]
]

**Comment by cosmiccoincidence** [comment]
To add to this:
The issue happens on any location with more than 1 stair tile per map tile. This happened to me in a prison where I went down a set of stairs and came back up to a different set of stairs, teleporting me.
I took a small video clip of me useing a survival shelter as an example, which has 4 stairs both on the main floor and basement. You can see that regardless of which set of stairs I go down, I always 'land' on the one in the bottom right corner. The same thing happens going up the stairs as well. And although I did not include it in the clip, I can confirm that peeking does it as well.
[video clip](https://cdn.discordapp.com/attachments/365907426270117888/679022823599112202/GotsdYZ7ar.mp4)

**Comment by I-am-Erk** [comment]
Unless this can be solved by saving location somehow before using stairs, the mapgen portion of this is too intractable to be a release blocker

**Comment by wapcaplet** [comment]
Retesting this in a build from today (0.D-12503-g14a261e7a1), I checked about 20 different houses and all of them are using the new matching upstairs/downstair maps with aligned staircases. None of the houses I saw had separate stairs in different parts of the house; they often had two staircases, but always adjacent.

The peek-to-teleport issue still exists, but is much less a concern when the stairs are directly adjacent. For example, here I peeked down the left stairs, and was teleported to the right stairs:

![image](https://user-images.githubusercontent.com/238395/75099400-78413100-557e-11ea-85d5-e80f955d4547.png)
